initial draft (#91)
fixie-ai/ultradox: guides/sip.mdx

---

> title: 'SIP Guide'

Reviewer: WDYT of adding "(BETA)" or something to the title?  Your warning hints at this, but it seems like something we should be very upfront about.

Author: done

Author: done

## SIP Quickstart

The fastest way to start using SIP with Ultravox:
<Steps>
  <Step title="Create an Agent">Create an agent using the [Ultravox Realtime console](https://app.ultravox.ai/agents).</Step>
  <Step title="Create a SIP Call">Create an agent call using the [Create Agent Call](/api-reference/agents/agents-calls-post) API and use the `sip` medium.</Step>
</Steps>

<Note>
  <b>API Key Required</b>
  <br />
  Make sure you have an Ultravox API key. You can create one in the [console](https://app.ultravox.ai/settings/).
</Note>

## SIP Overview

Ultravox uses a `medium` for each call, which defines how the call will occur. When using `sip` as the medium, you can connect your Ultravox agents with standard SIP-compatible systems.

You can create a SIP call with the Ultravox Realtime [Create Agent Call](/api-reference/agents/agents-calls-post) or [Create Call](/api-reference/calls/calls-post) endpoints using the `sip` medium.

The `sip` medium supports the creation of both incoming and outgoing calls.

### Incoming SIP Calls

> Incoming calls are initiated by a SIP client to your Ultravox agent. When the Ultravox incoming SIP call is created, we return a join URL that is then used to connect to the agent.

Reviewer: ```suggestion
Incoming calls are initiated by a SIP client to your Ultravox agent. When the Ultravox incoming SIP call is created, the returned join URL is a SIP URI to which an invite can be sent.
```

Author: great thx

### Outgoing SIP Calls

> Outgoing calls are when Ultravox initiates the call to a SIP endpoint or client. The configuration varies slightly depending on which SIP client/system you're using.

Reviewer: ```suggestion
Outgoing calls are when Ultravox initiates the call to a SIP endpoint or client. The configuration varies slightly depending on which SIP client/system you're using.
When creating an outbound SIP call, Ultravox will initiate the call directly and no join URL will be returned.
```

Author: done

## Create an Outgoing SIP Call

For outgoing SIP calls, you'll need to add the `outgoing` configuration data for the `sip` medium and add it to the request body of the [Create Agent Call](/api-reference/agents/agents-calls-post) or [Create Call](/api-reference/calls/calls-post) request:

```js
medium: {
  sip: {
    outgoing: {
      to: "sip:<destination>@<sip_domain>",
      from: "<your_identifier>",
      username: "<authentication_username>",
      password: "<authentication_password>"
    }
  }
}
```

### Linphone Example

```js Example: Creating an Outgoing SIP Call to Linphone
    medium: {
      sip: {
        outgoing: {
          to: "sip:<username>@sip.linphone.org",
          from: "<company_name>",
          username: "<linphone_username>",

>           password: "<linphone_password>"

Reviewer: Did you need these for Linphone?  I'm pretty sure it works without username/password since Linphone doesn't try to validate your caller id.

        }
      }
    }
    ```

## Create an Incoming SIP Call

For incoming SIP calls, you'll need to use `incoming` for the `sip` medium and add it to the request body of the [Create Agent Call](/api-reference/agents/agents-calls-post) or [Create Call](/api-reference/calls/calls-post) request. This will return a `joinUrl` that can be used to connect to a SIP client.

```js Example: Creating an Incoming SIP Call
medium: {
  sip: {
    incoming: {}
  }
}
```

## Supported Codecs
Ultravox Realtime currently supports the following codecs for SIP:

* PCMA
* PCMU

Using any other codec will cause calls to fail.

## Integrations

> [jambonz](https://jambonz.org) enables SIP connections and has an integration for Ultravox. [Learn more](/essentials/telephony#provider-specific-integration).

Reviewer: (optional) Sounds like we could mention voximplant now also

Author: done
